src/Symfony/Component/HttpKernel/Exception/HttpException.php

```diff
@@ -29,6 +29,27 @@ public function __construct(int $statusCode, string $message = '', \Throwable $p
         parent::__construct($message, $code, $previous);
     }
 
+    public static function fromStatusCode(int $statusCode, string $message = '', \Throwable $previous = null, array $headers = [], int $code = 0): self
+    {
+        return match ($statusCode) {
+            400 => new BadRequestHttpException($message, $previous, $code, $headers),
```

Comment:
Shouldn't the method signature follow an arguments order similar to exception?

Reply:
it follows the constructor instead, with provides local consistency (and the code is the less useful argument so it's better last to me)